server/dive_server/views_annotation.py

```diff
@@ -26,18 +27,33 @@ def __init__(self, resourceName):
         self.resourceName = resourceName
 
         self.route("GET", (), self.get_annotations)
+        self.route("GET", ("revision",), self.get_revisions)
         self.route("GET", ("export",), self.export)
 
         self.route("PATCH", (), self.save_annotations)
 
     @access.user
     @autoDescribeRoute(
-        Description("Get annotations of a clip").modelParam(
-            "folderId", **DatasetModelParam, level=AccessType.READ
-        )
+        Description("Get annotations of a dataset")
+        .pagingParams("trackId", defaultLimit=0)
+        .modelParam("folderId", **DatasetModelParam, level=AccessType.READ)
+        .param('revision', 'revision', dataType='integer', required=False)
     )
-    def get_annotations(self, folder):
-        return crud_annotation.get_annotations(folder)
+    def get_annotations(self, limit: int, offset: int, sort, folder, revision):
+        cursor, total = crud_annotation.get_annotations(folder, limit, offset, sort, revision)
+        cherrypy.response.headers['Girder-Total-Count'] = total
+        return cursor
+
+    @access.user
+    @autoDescribeRoute(
+        Description("Get dataset annotation revision log")
+        .pagingParams("revision", defaultLimit=20)
+        .modelParam("folderId", **DatasetModelParam, level=AccessType.READ)
+    )
+    def get_revisions(self, limit: int, offset: int, sort, folder):
```

Comment:
just wondering if the default sort order should be descending instead of ascending?  In actual usage may want the default to be the date/ltime/description of the last 20 instead of the first 20.

Reply:
I see that this was added to the `GET /dive_dataset` which is probably a good idea.  I meant for it to be added to the revisions here `GET /dive_annoation/revision`